src/libraries/System.Text.Json/tests/Serialization/CacheTests.cs

```diff
@@ -121,7 +121,7 @@ void Test(int i)
                 tasks[i + 1] = Task.Run(() => Test(TestClassCount - 2));
             }
 
-            Task.WaitAll(tasks);
+            await Task.WhenAll(tasks);
```

Comment:
FYI this area of code really does care about threading, and the changes shouldn't affect the multi-threading aspects.